pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -311,7 +311,7 @@ func (moa *MultiOrgAlertmanager) AlertmanagerFor(orgID int64) (*Alertmanager, er
 	}
 
 	if !orgAM.Ready() {
-		return nil, ErrAlertmanagerNotReady
+		return orgAM, ErrAlertmanagerNotReady
```

Comment:
@JohnnyQQQQ Pointing out the area where we're adding multiple error behaviors. Since `AlertmanagerFor` is used in many places, we discussed adding a separate method (say, `GetAlertmanager`) which doesn't check for readiness, and having `AlertmanagerFor` call our new method.

After experimenting with this, I ran into a couple issues of concern:

- `AlertmanagerFor` performs locking. If we push the lock acquisition and map access into our new method, we introduce an insidious potential bug where the AM readiness check no longer happens within the lock. Instead, it happens after the lock is released, allowing concurrent requests to interfere with state.
- To ensure both entry points are properly locked, we'd need them both to call a third private method which performs the lookup without any at all. Then, both entry points would independently handle their own locking. I didn't feel great about introducing a non-thread-safe private method, as it's easy for future developers to call this method without realizing it, introducing more concurrency problems.

The second error state here is not ideal, but IMO it is a "lesser of two evils." I really like the idea of having a single thread-safe lookup method so that callers don't have to think about which lookup method they're calling and whether it's safe or not.

Do you have any thoughts on this? I'm still happy to change this if we disagree with the reasoning above.